main.py

```diff
@@ -142,12 +242,34 @@ def api_template(my_path=''):
         if UPLOAD_LIMIT is not None and CURRENT_UPLOAD_SIZE >= UPLOAD_LIMIT:
             return Response(status=400, response='Reached upload limit')
         with open(os.path.join('static/data/', my_path,  'template.json'), "w+") as outfile:
+            data = request.get_json()
+        # Call the validate function from jsonschema to verify our output json conforms with the template we were supposed to create
+            try:
+                validate(data, TEMPLATE_SCHEMA)
+            except:
+                return Response(status=400, response='Invalid template, requires at least 1 label and proper formatting. Refer to schema.')
             json.dump(request.get_json(), outfile, sort_keys=True, indent=4)
+        # Open template file to make sure what we just created matches what was submitted. In theory we should never get an except here considering we JUST wrote the file. 
+        try:
+            template = open(os.path.join('static/data/',(my_path + 'template.json')))
+            # Convert the template json to something validate can understand.
+            template = json.load(template)
+        except:
```

Comment:
I think this block can be removed. We have at this point already verified the json and written it, just need to return success